wrapper/minecraft/src/main/kotlin/org/quiltmc/qkl/wrapper/minecraft/brigadier/ArgumentConstructor.kt

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qkl.wrapper.minecraft.brigadier
+
+import com.mojang.brigadier.arguments.ArgumentType
+import com.mojang.brigadier.builder.ArgumentBuilder
+import com.mojang.brigadier.builder.RequiredArgumentBuilder
+import com.mojang.brigadier.context.CommandContext
+
+public typealias RequiredArgumentConstructor<S, D> =
+        ArgumentConstructor<S, RequiredArgumentBuilder<S, *>, D>
+
+/**
+ * Class containing all data necessary to construct an
+ * [optional] or [required] argument.
+ */
+public class ArgumentConstructor<S, B : ArgumentBuilder<S, *>, D : ArgumentDescriptor<*>>(
+    private val builder: B,
+    private val name: String,
+    private val descriptor: D
+) {
+    /**
+     * Converts this constructor into a required argument.
+     *
+     * @see CommandArgument.Required
+     * @see required
+     *
+     * @author Cypher121
+     */
+    public fun required(): CommandArgument.Required<S, B, D> =
+        CommandArgument.Required(builder, name, descriptor)
```

Comment:
Make sure to avoid `fun =` syntax

Reply:
what's the reasoning behind that one actually? /gen

Reply:
Fix

Reply:
> what's the reasoning behind that one actually? /gen

Consistency